### PR TITLE
ECI-201 Add OCI metrics terraform setup

### PR DESCRIPTION
**Pull request by kanishktripathi**

### CODE REVIEW

File: datadog-oci-orm/metrics-setup/locals.tf [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+locals {
+  # OCI docker repository
+  oci_docker_repository = "${local.oci_region_key}.ocir.io/${local.ocir_namespace}"
+  oci_docker_host       = "${local.oci_region_key}.ocir.io"
+  ocir_repo_name        = "${var.resource_name_prefix}-functions"
+  function_name         = "datadog-function-metrics"
+  docker_image_path     = "${local.oci_docker_repository}/${local.ocir_repo_name}/${local.function_name}:latest"
+  # custom_image_path     = "${local.oci_region_key}.ocir.io/${var.function_image_path}"
```
Comment (bengillmandd) [inline datadog-oci-orm/metrics-setup/locals.tf:39]: Should this be removed?

File: datadog-oci-orm/metrics-setup/variables.tf [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  type        = string
+  description = "OCI Region as documented at https://docs.cloud.oracle.com/en-us/iaas/Content/General/Concepts/regions.htm"
+}
+variable "compartment_ocid" {
+  type        = string
+  description = "The compartment OCID to deploy resources to"
+}
+
+#*************************************
+#         TF auth Requirements for CLI
```
Comment (bengillmandd) [inline datadog-oci-orm/metrics-setup/variables.tf:86]: Is this kept here for documentation purposes?

**Review by bengillmandd — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,2 +1,17 @@
...
+# Oracle Cloud Integration
+Houses code for Datadog OCI integration. Includes code for:
+* OCI's log collections pipeline.
+* OCI Metric collection setup.
+
+
+
+## Deploy to OCI (metrics)
+
+The setup allows through terraform running on OCI resource manager (ORM) stack to:
```
Comment (bengillmandd) [inline README.md:10]: I believe there is a typo here

**Review by bengillmandd — APPROVED** [review]

File: datadog-oci-orm/metrics-setup/function_setup.tf [hunk 4]
```diff
@@ -0,0 +1,53 @@
+resource "null_resource" "Login2OCIR" {
+  count = local.user_image_provided ? 0 : 1
+  provisioner "local-exec" {
+    command = "echo '${var.oci_docker_password}' |  docker login ${local.oci_docker_repository} --username ${local.ocir_namespace}/${var.oci_docker_username} --password-stdin"
```
Comment (dtru-ddog) [inline datadog-oci-orm/metrics-setup/function_setup.tf:4]: this is fine, but have we considered using the docker TF resource instead?

File: datadog-oci-orm/metrics-setup/metrics-function/func.py
Comment (dtru-ddog) [inline datadog-oci-orm/metrics-setup/metrics-function/func.py:1]: Would be good to add some basic tests here.

**Review by dtru-ddog — APPROVED** [review]

**Pull request merged**